bool estop
float32[] values
